x-element.js

```diff
@@ -68,7 +87,9 @@ export default class XElement extends HTMLElement {
    * E.g., setup focus delegation or return host instead of host.shadowRoot.
    */
   static createRenderRoot(host) {
-    return host.attachShadow({ mode: 'open' });
+    const shadowRoot = host.attachShadow(this.shadowRootInit);
```

Comment:
I would recommend against the `shadowRootInit` concept — developers should simply override `createRenderRoot` to customize the render root init